vulcanus/src/main/java/com/merkury/vulcanus/features/user/UserEntityService.java

```diff
@@ -0,0 +1,21 @@
+package com.merkury.vulcanus.features.user;
+
+import com.merkury.vulcanus.model.entities.UserEntity;
+import com.merkury.vulcanus.model.repositories.UserEntityRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class UserEntityService {
+
+    private final UserEntityRepository userRepository;
+
+    public List<String> searchUsers(String username) {
+        List<UserEntity> users = userRepository.findTop10ByUsernameContainingIgnoreCase(username);
+
+        return users.stream().map(UserEntity::getUsername).toList();
+    }
```

Comment:
Skoro users jest używane tylko w jedym miejsu to możesz to odrazu do returna przekazać
```suggestion
     return userRepository
     .findTop10ByUsernameContainingIgnoreCase(username)
     .stream()
     .map(UserEntity::getUsername)
     .toList();
    }
```